control_panel_api/models.py

```diff
@@ -22,15 +26,33 @@ def get_short_name(self):
 
 
 class App(TimeStampedModel):
+    def _slugify(name):
+        """Create a slug using standard django slugify but we override with
+        extra replacement so it's valid for s3"""
+        return re.sub(r'_+', '-', slugify(name))
+
     name = models.CharField(max_length=100, blank=False)
-    slug = AutoSlugField(populate_from='name', slugify_function=services.app_slugify)
+    slug = AutoSlugField(populate_from='name', slugify_function=_slugify)
     repo_url = models.URLField(max_length=512, blank=True, default='')
 
     class Meta:
         ordering = ('name',)
 
+    @property
+    def aws_role_name(self):
+        return f"{settings.ENV}_app_{self.slug}"
+
+    def aws_create_role(self):
+        services.create_app_role(self.aws_role_name)
+
+    def aws_delete_role(self):
+        services.delete_app_role(self.aws_role_name)
+
 
 class S3Bucket(TimeStampedModel):
+    READWRITE = 'readwrite'
```

Comment:
Two things:
1. This may be misleading here, S3 buckets are not RO/RW per se
2. It doesn't look like we actually use these constants anywhere?

Reply:
Well spotted - was old code from when I was trying something else.